ledger/participant-integration-api/src/main/scala/platform/apiserver/error/LedgerApiErrors.scala

```diff
@@ -0,0 +1,506 @@
+          err: LfError.Preprocessing.Error
+      )(implicit
+          loggingContext: LoggingContext,
+          logger: ContextualizedLogger,
+          correlationId: CorrelationId,
+      ) extends LoggingTransactionErrorImpl(
+            cause = err.message
+          )
+    }
+  }
+
+  object InterpreterErrors extends ErrorGroup {
+
+    @Explanation("""This error occurs if the Daml transaction failed during interpretation.""")
+    @Resolution("This error type occurs if there is an application error.")
+    object GenericInterpretationError
+        extends ErrorCode(
+          id = "DAML_INTERPRETATION_ERROR",
+          // TODO error codes: this is a bad error message and needs to be fixed by also adjusting the expected ledger-api conformance tests
+          ErrorCategory.InvalidIndependentOfSystemState, // (is ILLEGAL_ARGUMENT, should be PRECONDITION_FAILED)
+        ) {
+
+      case class Error(override val cause: String)(implicit
+          loggingContext: LoggingContext,
+          logger: ContextualizedLogger,
+          correlationId: CorrelationId,
+      ) extends LoggingTransactionErrorImpl(
+            cause = cause
+          )
+
+    }
+
+    @Explanation(
+      """This error occurs if an exercise or fetch happens on a transaction-locally consumed contract."""
+    )
+    @Resolution("This error indicates an application error.")
+    object ContractNotActive
+    // Is INVALID_ARGUMENT, SHOULD BE NOT_EXISTS
+        extends ErrorCode(
+          id = "CONTRACT_NOT_ACTIVE",
+          ErrorCategory.InvalidIndependentOfSystemState,
+        ) {
+
+      case class Reject(
+          override val cause: String,
+          _err: LfInterpretationError.ContractNotActive,
+      )(implicit
+          loggingContext: LoggingContext,
+          logger: ContextualizedLogger,
+          correlationId: CorrelationId,
+      ) extends LoggingTransactionErrorImpl(
+            cause = cause
+          ) {
+        override def resources: Seq[(ErrorResource, String)] = Seq(
+          (ErrorResource.ContractId, _err.coid.coid)
+        )
+      }
+
+    }
+
+    @Explanation(
+      """This error signals that within the transaction we got to a point where two contracts with the same key were active."""
+    )
+    @Resolution("This error indicates an application error.")
+    object DuplicateContractKey
+        extends ErrorCode(
+          id = "DUPLICATE_CONTRACT_KEY_DURING_INTERPRETATION",
+          ErrorCategory.InvalidGivenCurrentSystemStateResourceExists,
+        ) {
+      case class Reject(
+          override val cause: String,
+          _key: GlobalKey,
+      )(implicit
+          loggingContext: LoggingContext,
+          logger: ContextualizedLogger,
+          correlationId: CorrelationId,
+      ) extends LoggingTransactionErrorImpl(
+            cause = cause
+          ) {
+        override def resources: Seq[(ErrorResource, String)] = Seq(
+          (ErrorResource.ContractKey, _key.toString())
```

Comment:
this was one of the places where I was not sure: what should be the format to transport a contract key to the user.

Reply:
Added a comment